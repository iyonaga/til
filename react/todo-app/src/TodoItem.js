import React, { Component } from 'react';

export default class TodoItem extends Component {
  render() {
    const { todo, i, isCompleted, handleChange } = this.props;
    return (
      <li>
        <input type="checkbox" onChange={handleChange} value={i} checked={isCompleted} />
          { isCompleted ? <s>{todo.title}</s> : todo.title }
      </li>
    )
  }
}
