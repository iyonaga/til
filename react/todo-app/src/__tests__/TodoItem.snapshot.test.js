import React from 'react';
import TodoItem from '../TodoItem';
import renderer from 'react-test-renderer';

describe('TodoItem', () => {
  it('TodoItem component renders the todo correctly', () => {
    const todo = {
      title: 'dummy',
      status: 'active'
    }

    const tree = renderer.create(
      <TodoItem todo={todo} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })
})
