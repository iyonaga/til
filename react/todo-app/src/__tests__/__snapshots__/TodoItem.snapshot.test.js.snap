// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem TodoItem component renders the todo correctly 1`] = `
<li>
  <input
    checked={undefined}
    onChange={undefined}
    type="checkbox"
    value={undefined}
  />
  dummy
</li>
`;
